backend/models/core.go

```diff
@@ -0,0 +1,57 @@
+package models
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/dataframe"
+)
+
+// PluginConfig holds configuration for the queried plugin.
+type PluginConfig struct {
+	ID       int64
+	OrgID    int64
+	Name     string
+	Type     string
+	URL      string
+	JSONData json.RawMessage
+}
+
+type DataQueryRequest struct {
+	PluginConfig PluginConfig
+	Headers      map[string]string
+	Queries      []DataQuery
+}
+
+// DataQuery represents the query as sent from the frontend.
+type DataQuery struct {
+	RefID         string
+	MaxDataPoints int64
+	Interval      time.Duration
+	TimeRange     TimeRange
+	JSON          json.RawMessage
+}
+
+// DataQueryResponse holds the results for a given query.
+type DataQueryResponse struct {
+	Frames   []*dataframe.Frame
+	Metadata map[string]string
```

Comment:
What goes in this metadata?  on the frontend, we don't have anywhere for it to go... 

(each Frame has metadata though)

Reply:
I think @kylebrandt have fixed this in another PR or else we need to merge in master?

Reply:
This has been there since @ryantxu original proposed major changes: https://github.com/grafana/grafana-plugin-sdk-go/pull/21/files#diff-ad0dcffdf701a0711096c6d383ad6cd5R82 : 

```
message QueryResponse {
  // Arrow encoded DataFrames
  // Each frame encodes its own: Errors, meta, and refId
  repeated bytes data = 1;

  // Additional response metadata
  map<string,string> headers = 2;
}
```

I assumed this was a place for metadata about the entire response (that contains multiple data frames). In this sense I am guess I am confused about multiple "QueryResultMeta" objects being in a single QueryResponse which was added in #29. 

Seems like maybe an separate issue and not part of this PR?